Source/NSMenuToolbarItem.m

```diff
@@ -0,0 +1,90 @@
+/* Implementation of class NSMenuToolbarItem
+   Copyright (C) 2024 Free Software Foundation, Inc.
+
+   By: Gregory John Casamento
+   Date: 03-05-2024
+
+   This file is part of the GNUstep Library.
+
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110 USA.
+*/
+
+#import "AppKit/NSMenuToolbarItem.h"
+#import "AppKit/NSMenu.h"
+#import "AppKit/NSImage.h"
+
+#import "GNUstepGUI/GSTheme.h"
+
+@interface NSToolbarItem (Private)
+- (NSView *) _backView;
+@end
+
+@implementation NSMenuToolbarItem
+
+- (instancetype) initWithItemIdentifier: (NSString *)identifier
+{
+  self = [super initWithItemIdentifier: identifier];
+  if (self != nil)
+    {
+      [self setImage: [NSImage imageNamed: @"NSMenuToolbarItem"]];
+      [self setTarget: self];
+      [self setAction: @selector(_showMenu:)];
+    }
+  return self;
+}
+
+- (void) dealloc
+{
+  RELEASE(_menu);
+  [super dealloc];
+}
+
+- (BOOL) showsIndicator
+{
+  return _showsIndicator;
+}
+
+- (void) setShowsIndicator: (BOOL)flag
+{
+  _showsIndicator = flag;
+
+  if (_showsIndicator == YES)
+    {
+      [self setImage: [NSImage imageNamed: @"NSMenuToolbarItem"]];
+    }
+  else
+    {
+      [self setImage: nil];
+    }
+}
+
+- (NSMenu *) menu
+{
+  return _menu;
+}
+
+- (void) setMenu: (NSMenu *)menu
+{
+  ASSIGN(_menu, menu);
+}
+
+- (void) _showMenu: (id)sender
+{
+  [[GSTheme theme] rightMouseDisplay: _menu
```

Comment:
Where is this menu being displayed? Should we fake an event here to pass a location or are there better ways to control that?

Reply:
The menu shows when you right-click on the NSToolbarItem icon that represents it.

Reply:
Here is an example on GNUstep:

![menu_item_toolbar_test](https://github.com/gnustep/libs-gui/assets/27943/e1883788-d051-45dc-8c4c-baa9ab06343e)

And on the mac:

![mac_menu_toolbar_item](https://github.com/gnustep/libs-gui/assets/27943/a4c975b0-3830-45c3-a0c5-9cef8dae5d4a)

Both are using the test code referenced in the previous comments.